feat: Add an action to check sentry-cli is installed

matt-oakes opened the pull request:
Added a new action which will only pass if the sentry-cli is installed and meets the minimum version.

This is intended to be used at the start of the lane to ensure the machine is setup correctly before preceding to build and upload the app. This avoids long waits just to see that the Sentry upload will fail later on, which might leave some builds being uploaded to the App Store without source maps also being uploaded.

philipphofmann commented on lib/fastlane/plugin/sentry/actions/sentry_check_cli.rb:
```diff
@@ -0,0 +1,37 @@
+      #####################################################
+
+      def self.description
+        "Checks to make sure that sentry-cli is installed and at the correct version"
```
`l`
```suggestion
        "Checks that sentry-cli with the correct version is installed"
```

philipphofmann commented on lib/fastlane/plugin/sentry/actions/sentry_check_cli.rb:
```diff
@@ -0,0 +1,37 @@
+      def self.details
+        [
+          "This action checks that the senty-cli is installed and meets the mimum verson requirements.",
+          "It is recommended that you run this at the start of your lane to make sure that sentry-cli is correctly setup for later in the lane."
```
```suggestion
          "You can use it at the start of your lane to ensure that sentry-cli is correctly installed."
```

philipphofmann commented on README.md:
````diff
@@ -118,6 +118,14 @@ sentry_set_commits(
 )
 ```
 
+#### Checking the sentry-cli is setup
````
```suggestion
#### Checking the sentry-cli is installed
```

philipphofmann commented on README.md:
````diff
@@ -118,6 +118,14 @@ sentry_set_commits(
+Useful for checking that the sentry-cli is installed and meets the mimum version requirements before starting to build your app in your lane.
+
+```ruby
+sentry_check_cli()
````
What do you think about renaming this to?
```suggestion
sentry_check_cli_installed()
```

philipphofmann reviewed (commented):
@matt-oakes, thanks a lot for this PR 😀. This is pretty useful. I made a few minor suggestions. Can you also please add tests for this. Then we should be good to merge. Sorry for the late reply.

matt-oakes commented:
@philipphofmann Thanks for the review. I've made the requested changes and renamed the command. I've added a test fo the "success" case, but I don't know enough about Ruby and Rspec to add a test for the failure case. I presume I need to mock the response to `which sentry-cli`, but I have no idea how I'd do that. Feel free to add this in yourself, if needed.

Also, I was getting an issue where the tests were failing on `master`. It turned out to be the `sentry_config_spec.rb` test was failing because I had a `.sentryclirc` file in my home folder, which was up the tree from where the tests were being run. I presume that something needs to be done here to isolate the test environment a little more.

philipphofmann commented:
Sorry again for the late review, @matt-oakes. Can you please add an entry to the changelog?

matt-oakes commented:
@philipphofmann Sorry for the delay. Rebased and added the changelog entry.

philipphofmann reviewed (approved):
Good job 👏, thanks @matt-oakes.
